### PR TITLE
Eos 27080,Eos 27082 - LR data table changes.

### PR DESCRIPTION
**Pull request by sujit-h-singh**
# Problem Statement
- LR data table which would be used to display data in a tabular format. Consist of features as custom column, date column, hover actions, search and advance search.

# Design
-  [Confluence-design](https://seagate-systems.atlassian.net/l/c/Zk1Qn3nC)
- [Figma design](https://www.figma.com/file/HuYNdGvVazilCkD8mJO1qQ/LR-Alert-page-redesign-options?node-id=292%3A63836)

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: gui/src/components/widgets/LrDataTable/LrDataTable.constant.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const lrDataTableConst = {
```
Comment (NareshGopal) [inline gui/src/components/widgets/LrDataTable/LrDataTable.constant.ts:1]: Add the license comment

File: gui/tslint.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
     "max-classes-per-file": [
       false,
       1
-    ]
+    ],
+    "curly": [true, "ignore-same-line"]
```
Comment (NareshGopal) [inline gui/tslint.json:41]: Can we remove the "ignore-same-line" flag? I feel the code to be more readable when we add curly braces for control statements.

**Review by NareshGopal — CHANGES_REQUESTED** [review]

File: gui/src/components/widgets/LrDataTable/LrDataTable.constant.ts (hunk 1 above)
Comment (sujit-h-singh) [reply gui/src/components/widgets/LrDataTable/LrDataTable.constant.ts:17]: added comments at missing places.

File: gui/tslint.json (hunk 2 above)
Comment (sujit-h-singh) [reply gui/tslint.json:41]: This is now a very standard practice of not adding curly braces when you have only one line code for control statement.

File: gui/src/components/widgets/LrDataTable/LrDataTable.constant.ts (hunk 1 above)
Comment (NareshGopal) [reply gui/src/components/widgets/LrDataTable/LrDataTable.constant.ts:17]: Okay

File: gui/tslint.json (hunk 2 above)
Comment (NareshGopal) [reply gui/tslint.json:41]: Okay

**Review by NareshGopal — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**